src/main/java/org/prebid/server/bidder/krushmedia/KrushmediaBidder.java

```diff
@@ -0,0 +1,172 @@
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.krushmedia.ExtImpKrushmedia;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class KrushmediaBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpKrushmedia>> KRUSHMEDIA_EXT_TYPE_REFERENCE =
+            new TypeReference<ExtPrebid<?, ExtImpKrushmedia>>() {
+            };
+    private static final String URI_ACCOUNT_ID_MACRO = "{{AccountID}}";
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public KrushmediaBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<BidderError> errors = new ArrayList<>();
+
+        final ExtImpKrushmedia extImpKrushmedia;
+        final String url;
+
+        try {
+            extImpKrushmedia = parseImpExt(request.getImp().get(0));
+            url = resolveEndpoint(extImpKrushmedia.getAccountId());
+        } catch (PreBidException e) {
+            return Result.emptyWithError(BidderError.badInput(e.getMessage()));
+        }
+
+        final BidRequest outgoingRequest = request.toBuilder()
+                .imp(removeFirstImpExt(request.getImp()))
+                .build();
+
+        return Result.of(Collections.singletonList(
+                HttpRequest.<BidRequest>builder()
+                        .method(HttpMethod.POST)
+                        .uri(url)
+                        .headers(resolveHeaders(request.getDevice()))
+                        .payload(outgoingRequest)
+                        .body(mapper.encode(outgoingRequest))
+                        .build()),
+                errors);
+    }
+
+    private ExtImpKrushmedia parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), KRUSHMEDIA_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException("Error while unmarshalling bidder extension");
+        }
+    }
+
+    private String resolveEndpoint(String accountId) {
+        return endpointUrl.replace(URI_ACCOUNT_ID_MACRO, accountId);
```

Comment:
i'd rather suggest to wrap `accountId` with `StringUtils.stripToEmpty(..)` to avoid `=null` param value in URL.